homeassistant/components/upnp/config_flow.py

```diff
@@ -32,8 +32,8 @@
 def _friendly_name_from_discovery(discovery_info: ssdp.SsdpServiceInfo) -> str:
     """Extract user-friendly name from discovery."""
     return (
-        discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME)
-        or discovery_info.upnp.get(ssdp.ATTR_UPNP_MODEL_NAME)
+        cast(str, discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME))
```

Comment:
This'll do for now, but perhaps `ssdp._UpnpServiceDescription` can be better defined (future work).

Reply:
I have adjusted this